src/tool/xmeta/cpp/ast_to_st_listener.cpp

```diff
@@ -0,0 +1,206 @@
+#include <algorithm>
+#include <stdexcept>
+#include "symbol_table.h"
+#include "models/xmeta_models.h"
+#include "ast_to_st_listener.h"
+
+using namespace xlang::xmeta;
+extern bool semantic_error_exists;
+
+std::map<std::string, enum_semantics> enum_semantics_map
```

Comment:
make this static and const?

Reply:
Even worse, why is this a global object. If we need to perform this mapping, hide the details behind a function.